rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -318,17 +393,58 @@ private void commitProposedFederation(ActivationConfig.ForBlock activations) {
         var blockNumber = 
-            initialBlock.getNumber() + BRIDGE_CONSTANTS.getFederationConstants().getFederationActivationAge(activations);
-        var initialBlockHeader = new BlockHeaderBuilder(mock(ActivationConfig.class))
+            currentBlock.getNumber() + BRIDGE_CONSTANTS.getFederationConstants().getFederationActivationAge(activations);
+        var blockHeader = new BlockHeaderBuilder(mock(ActivationConfig.class))
             .setNumber(blockNumber)
             .build();
-        var activationBlock = Block.createBlockFromHeader(initialBlockHeader, true);
+        currentBlock = Block.createBlockFromHeader(blockHeader, true);
 
         // Now the new bridgeSupport points to the new block where the new federation
         // is considered to be active
-        bridgeSupport = getBridgeSupportFromExecutionBlock(activationBlock, activations);
+        bridgeSupport = getBridgeSupportFromExecutionBlock(currentBlock, activations);
     }
-    
+
+    private void activateMigration(ActivationConfig.ForBlock activations) {
+        // Move the required blocks ahead for the new federation to start migrating,
+        // adding 1 as the migration is exclusive
+        var blockNumber = 
+            currentBlock.getNumber() + BRIDGE_CONSTANTS.getFederationConstants().getFundsMigrationAgeSinceActivationBegin() + 1L;
+        var blockHeader = new BlockHeaderBuilder(mock(ActivationConfig.class))
+            .setNumber(blockNumber)
+            .build();
+        currentBlock = Block.createBlockFromHeader(blockHeader, true);
+
+        bridgeSupport = getBridgeSupportFromExecutionBlock(currentBlock, activations);
+    }
+
+    private void endMigration(ActivationConfig.ForBlock activations) throws Exception {
+        // Move the required blocks ahead for the new federation to finish migrating,
+        // adding 1 as the migration is exclusive
+        var blockNumber = 
+            currentBlock.getNumber() + BRIDGE_CONSTANTS.getFederationConstants().getFundsMigrationAgeSinceActivationEnd(activations) + 1L;
+        var blockHeader = new BlockHeaderBuilder(mock(ActivationConfig.class))
+            .setNumber(blockNumber)
+            .build();
+        currentBlock = Block.createBlockFromHeader(blockHeader, true);
+
+        bridgeSupport = getBridgeSupportFromExecutionBlock(currentBlock, activations);
+
+        // The first update collections after the migration finished should get rid of the retiring powpeg
+        var updateCollectionsTx = buildUpdateCollectionsTx();
+        bridgeSupport.updateCollections(updateCollectionsTx);
+        bridgeSupport.save();
+    }
+
+    private void migrateUTXOs() throws Exception {
+        // Migrate while there are still utxos to migrate
+        var remainingUTXOs = federationStorageProvider.getOldFederationBtcUTXOs();
+        while (!remainingUTXOs.isEmpty()) {
+            var updateCollectionsTx = buildUpdateCollectionsTx();
```

Comment:
```suggestion
        // Migrate while there are still utxos to migrate
        var remainingUTXOs = federationStorageProvider.getOldFederationBtcUTXOs();
        var updateCollectionsTx = buildUpdateCollectionsTx();
        while (!remainingUTXOs.isEmpty()) {
```

it always created the same tx right?

Reply:
Nice catch , thanks!